unified-prototype/unified-plugin/plugin-kmp/src/main/java/org/gradle/api/experimental/kmp/StandaloneKmpLibraryPlugin.java

```diff
@@ -15,18 +17,37 @@
 public class StandaloneKmpLibraryPlugin implements Plugin<Project> {
     @Override
     public void apply(Project project) {
-         KmpLibrary dslModel =project.getExtensions().create("kmpLibrary", AbstractKmpLibrary.class);
+        KmpLibrary dslModel = createDslModel(project);
 
         // Register an afterEvaluate listener before we apply the Android plugin to ensure we can
         // run actions before Android does.
         project.afterEvaluate(p -> linkDslModelToPlugin(p, dslModel));
 
-        // Apply the official KMP plugin.
+        // Apply the official KMP plugin
         project.getPlugins().apply("org.jetbrains.kotlin.multiplatform");
 
         linkDslModelToPluginLazy(project, dslModel);
     }
 
+    private KmpLibrary createDslModel(Project project) {
+        KmpLibrary dslModel = project.getExtensions().create("kmpLibrary", AbstractKmpLibrary.class);
+
+        // In order for function extraction from the DependencyCollector on the library deps to work, configurations must exist
+        // Matching the names of the getters on LibraryDependencies
+        DependencyScopeConfiguration api = project.getConfigurations().dependencyScope("api").get();
```

Comment:
So we don't _need_ to wire the dependency collector to these configurations, so this code is extraneous:

```
project.getConfigurations().getByName("api").fromDependencyCollector(dslModel.getDependencies().getApi());
project.getConfigurations().getByName("implementation").fromDependencyCollector(dslModel.getDependencies().getImplementation());
```

We _do_ use `linkSourceSetToDependencies` in `linkDslModelToPluginLazy` to connect these `DependencyCollector`s to the `commonMain` sourceSet.

But we still do need to _create_ these configurations in order for the function extractor to work, since it detects getter methods returning `DependencyCollector` that match the name of existing configurations.  So if we want to call:

```
kmpLibrary {
	dependencies {
	    implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.4.1")
	}
}
```

we need a `implementation` configuration, etc.  The extraneous wiring code does allows these deps to show up in the `dependencies` report, which I had been using for testing, but I'll remove now.

Reply:
Why does the function extractor code need to care about configurations at all? 

Imo the function extractor code for a dependency collector should be independent of configurations

Reply:
I was thinking to avoid false positives, if someone writes a `DC` valued getter somewhere that's not meant to be wired to a configuration, it's just making the `DC` available for something else to use.